### ISSUE TITLE
Joining on scalars rather than objects

### ISSUE THREAD

**Issue by kamek-pf** [join-monster/join-monster#57]
Hi !

So here is my use case, I'm dealing with some fields that need to be translated in my database :

some_table |
------------ |
id `int` |
some_value `int` |
... |

some_table_i18n |
------------ |
id `int references some_table (id)` |
language `varchar(2) ...`
some_string `varchar(100)`
... |

`some_table` holds all standard fields, `some_table_i18n` holds all fields that need to be translated, it has a reference to `some_table`.
Pretty standard way to handle translations in a relational database. My GraphQL schema is based on `some_table`.

My GraphQL API should let me do two things :
- ask for the `someString` field and get a value translated in the user's language (by looking at HTTP headers)
- ask for a `someStringIntl` field, an object that can return the different values of the string (this field will be used by a CMS)

A query could look like this :
```
...
{
    someTable {
       someValue // -> GraphQLInt
       someString // -> GraphQLString
       someStringIntl {
           en // -> GraphQLString
           fr // -> GraphQLString
       }
}
...
```

Maybe I missed something, but it looks like Join Monster does not support joins on scalars : I can't use `sqlJoin` if the type of my field is not `GraphQLObjectType`, and in the case of `someString`, it isn't (as it is just a string).

For now, I'm cheating with a view that simply merges both tables based on IDs. Doing so, everything looks like it's already in the table from Join Monster's point of view.
This resolves my first point, but I'm kind of stuck on the other.

Any idea how to tackle this ? :)

**Comment by acarl005** [comment]
Hello @kamek-pf. I like learning about these use cases. It certainly helps me improve the project.

This stumped me for a while. I would've taken the same approach for your first point. We've found that relying on views rather than the actual tables is the best way to accommodate SQL schemas.

To your second point, what if you made a slight change to the schema...
```graphql
{
    someTable {
       someValue # -> GraphQLInt
       someString # -> GraphQLString
       someStringIntl(language: "en") {
           someString # -> GraphQLString
       }
}
```

By making the language a parameter rather than a field, it might work. You could put the language into the where condition. Or better yet, I could add another parameter to the `sqlJoin` function which receives the GraphQL args. Then the `JOIN` condition could include the language.

```graphql
{
    someTable {
       someValue # -> GraphQLInt
       someString # -> GraphQLString
       en: someStringIntl(language: "en") {
           someString # -> GraphQLString
       }
       fr: someStringIntl(language: "fr") {
           someString # -> GraphQLString
       }
}
```

You could call the `someStringIntl` field multiple times using GraphQL aliases.

Would this fulfill your use case?

P.S. It looks like GitHub has added flavored markdown support for GraphQL!

**Comment by kamek-pf** [comment]
Hi @acarl005,

I spent some time thinking about all this and I came up with similar ideas.

I don't think being implicit about the language (like I wanted to with `someString`) is such a good idea anymore. If the client application needs to let the user choose a different language, then a GraphQL client lib like Relay would have no way to tell what fields need to change.

I'll probably go for a new `TranslatedString` object type :
```graphql
TranslatedString(lang: String):
    language: String,
    text: String
```
Every string that has to be translated could be an array of `TranslatedString`. It could be filtered with the `lang` parameter, or return the whole list if the parameter isn't specified.

```graphql
{
    someTable {
        someString(lang: "en") { text } # as a regular client
        someString { language, text } # as a CMS user
    }
}
```
This would cover all my use cases. 
`someString(lang: "en") { text }` would always return an array with exactly one element, a bit awkward, but  still a decent trade off.

This can already be implemented in the current state of Join Monster (by filtering from within the resolver), but as you said, having the args accessible from `sqlJoin` would be even better. I imagine this could be useful  in many different situations.

Thanks again for taking the time to look at this @acarl005 :D
